models/intermediate/int_asana__task_projects.sql

```diff
@@ -1,57 +1,87 @@
 with task_project as (
 
-    select * 
+    select *
     from {{ var('project_task') }}
 
 ),
 
 project as (
-    
-    select * 
+
+    select *
     from {{ var ('project') }}
 ),
 
 task_section as (
 
-    select * 
+    select *
     from {{ var('task_section') }}
 
 ),
 
 section as (
-    
-    select * 
+
+    select *
     from {{ var ('section') }}
 
 ),
 
 task_project_section as (
 
-    select 
+    select
         task_project.task_id,
         project.project_name || (case when section.section_name = '(no section)' then ''
-            else ': ' || section.section_name end) as project_section, 
-        project.project_id
+            else ': ' || section.section_name end) as project_section,
+        cast(project.project_id as {{ dbt.type_string() }}) as project_id,
+        project.project_name,
+        -- Using the first detected task project pairing as the "primary"
+        row_number() over (partition by task_project.task_id 
+            order by task_project._fivetran_synced asc, project.created_at asc) = 1 as is_first_project
     from
-    task_project 
-    join project 
+    task_project
+    join project
         on project.project_id = task_project.project_id
     join task_section
         on task_section.task_id = task_project.task_id
-    join section 
-        on section.section_id = task_section.section_id 
+    join section
+        on section.section_id = task_section.section_id
         and section.project_id = project.project_id
 ),
 
+task_project_primary as (
+    select distinct
+        task_id,
+        task_project_section.project_id as first_project_id,
+        task_project_section.project_name as first_project_name
+    from task_project_section
+    where is_first_project
```

Comment:
I am a bit wary of this cte. I was checking our own internal data and when an initial sync takes place we notice that all task project records have the same sync time per `_fivetran_synced`. See the below query and results. As such, I am unsure which of the following are in fact the first task project. 

![image](https://github.com/fivetran/dbt_asana/assets/74217849/f6093c6d-d3b8-426b-8002-84b5178da582)


Additionally, I don't know how confident I feel about the window function. We are selecting the first project based off the sync date (which as we see above could be the same for multiple projects) and the created_date. However, the created_date of the project may not relate to when the task was associated with the project. I worry we could be providing incorrect assumptions regarding what the first project of a task is with this approach.

What are your thoughts? I would almost lean towards removing these `first_project_*` fields as we cannot 100% validate that they are accurate. Ideally, we would be able to recreate a similar [jira__daily_issue_field_history](https://fivetran.github.io/dbt_jira/#!/model/model.jira.jira__daily_issue_field_history) model to truly see the changing attributes of the task within Asana (and therefore see which project was first/last/etc.). However, since this is not available in the Asana ERD, we cannot recreate.

Reply:
Hey @fivetran-joemarkiewicz thank you for the feedback!!  Yeah i was struggling with this same thought, which is that we can't 100% reliably know which project a task was added to "first", but yet that firstness has some business value.  But until there's some combination of Asana API improvements and Fivetran sync improvements, we've got what we've got, right? 🙂 

So I thought, what if we treat this as a "mostly" reliable field? I'm not sure if that would work for all your customers (or could be effectively communicated to them), but it provided me some value so I thought I'd pass it along.  But also I'd be understanding if you didn't want to ship the `first_` fields due to customer perception issues.

And yeah I totally understand how if you had to force resync your Asana data, or just started syncing an instance with a long history, this field would be mostly incorrect for you (unless we could think of a better tie-breaker than `created_date`?)